CustomEventEmitter.js

```diff
@@ -0,0 +1,100 @@
+class EventEmitter {
+  listeners = {}; // key-value pair
+
+  addListener(eventName, fn) {
+    if (this.listeners[eventName]) {
+      this.listeners[eventName].add(fn);
+    } else {
+      this.listeners[eventName] = new Set();
+      this.listeners[eventName].add(fn);
+    }
+  }
+
+  on(eventName, fn) {
+    this.addListener(eventName, fn);
+  }
+
+  removeListener(eventName, fn) {
+    if (this.listeners[eventName]) {
+      this.listeners[eventName].delete(fn);
+    }
+  }
+
+  off(eventName, fn) {
+    this.removeListener(eventName, fn);
+  }
```

Comment:
But it is not same for the `off` and `removeListener`. Actually `off` is an alias for `removeListener` . Kinda weird :) But it is as it is.